pkg/terraform/stages/vsphere/stages.go

```diff
@@ -0,0 +1,13 @@
+package vsphere
+
+import (
+	"github.com/openshift/installer/pkg/terraform"
+	"github.com/openshift/installer/pkg/terraform/stages"
+)
+
+// PlatformStages are the stages to run to provision the infrastructure in vsphere.
+var PlatformStages = []terraform.Stage{
+	stages.NewStage("vsphere", "pre-bootstrap"),
+	stages.NewStage("vsphere", "bootstrap", stages.WithNormalDestroy()),
+	stages.NewStage("vsphere", "master"),
```

Comment:
Is there a reason to use three stages here? Can we not use "Cluster" and then "Bootstrap" where cluster is the combination of pre-bootstrap and master? It's entirely possible to deploy the master instances before we deploy the bootstrap host as they will simply be waiting on ignition until bootstrap is up. I'd prefer consistency... and one of these is not like the others.

- https://github.com/openshift/installer/pull/5010/commits/5c9e55ca35de20ee5e71264fc72e367036f36ee6#diff-0bbc45f383d06f3383bae5fb8e48e47f2ffa3182726b922e6e66321e93c55a5dR10-R11
- https://github.com/openshift/installer/pull/5010/commits/05e59bb8082491eff09ba594c73f42dc7bf27be5#diff-160e82277dbca268921ba3315fc07e8abac5e7bc6c5a94527ab1b92fa2d11599R13-R15
- https://github.com/openshift/installer/pull/5119/files#diff-d7310ec3465d4577b8f70742e7ecd5a22f463698d1aa807da17041613bcd486eR9-R11

Reply:
The idea was to have the first stage which was a prerequisite (template) for both the bootstrap and master stages.
It sounds like a good idea to standardize. Thanks for the input!

Reply:
I like @jstuever's suggestion. Azure has a limitation that master VMs will not create successfully until they have pulled ignition (this is different than other platforms) from bootstrap. So in the case of Azure I think it makes sense to create masters in a separate stage after bootstrap, but we should only do that as needed.

Reply:
@jstuever @patrickdillon I get a similar result to Azure by combining the 2 stages as a first "Cluster" stage, when using vSphere + output variables since we require the vmware tools on the vm to be first installed, before we can gather data in the output variables (bootstrap has to spin up and mcs, then the masters have to ignite and then reboot). 

I confirmed that the actual result of doing that in the suggested order is that the control nodes will wait for ignition until they time out with `"Error: timeout waiting for an available IP address"`

Reply:
@jcpowermac I think your input could be useful here as well!

Reply:
- open-vm-tools is already installed in the image, just isn't started until after ignition
- The length of time for masters to be built, ignited might extend beyond the allocated time to grab the IP from tools - which is my concern with this.

Reply:
Thanks! What I've seen is an infinite loop on the masters trying to fetch the ignition: `ignition: GET error: Get "https://$apiVIP/22623/config/master" dial tcp $apiVIP:22623: connect: no route to host` from the control nodes console, and the output variable for the control node IP address timing out (even after extending the timeouts to 45 minutes), the control node's IP address could not be found through this process.

Reply:
I rolled back to 3 stages, and I was not able to reproduce the timeouts for the IP addresses with vmware tools so unless we hit this issue (I believe it was of my own) again, we should be able to roll use the stages design for vsphere.